libs/dal/src/repositories/integration/integration.repository.ts

```diff
@@ -28,4 +29,35 @@ export class IntegrationRepository extends BaseRepository<IntegrationEntity> {
       'Duplicate key - One application may not have two providers of the same channel type'
     );
   }
+
+  async update(
+    query: FilterQuery<IntegrationEntity & Document>,
+    updateBody: any
+  ): Promise<{
+    matched: number;
+    modified: number;
+  }> {
+    if (query.active === true) {
+      await this.deactivatedOtherActiveChannels(query);
+    }
+
+    return await super.update({ _id: query._id }, updateBody);
+  }
+
+  async deactivatedOtherActiveChannels(query: FilterQuery<IntegrationEntity & Document>): Promise<void> {
+    const otherExistedIntegration = (
+      await this.find({
+        _applicationId: query._applicationId,
+        channel: query.channel,
+        active: true,
+      })
+    ).filter((x) => x.providerId !== query.providerId);
```

Comment:
```suggestion
    const otherExistedIntegration = (
      await this.find({
         _id: {$ne: query._id},
         providerId: query.providerId,
        _applicationId: query._applicationId,
        channel: query.channel,
        active: true,
      })
    );
```

This will move the filtering to the DB